### ISSUE TITLE
problem with psf or dcd in VMD GUI

### ISSUE THREAD

**Issue by chiendarret** [prody/ProDy#1215]
I tried to install prody via conda (Anaconda3 on Debian buster. After about four hours software conflicts are being examined I wonder whether there is any chance for a successful installation.

Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: | 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
Examining conflict for dask h5py pytables biopython matplotl

**Comment by chiendarret** [comment]
Overnight finished revealing a large number of conflicts, se attached files. Updating Anaconda?
![Screenshot_2020-12-12_01-49-00](https://user-images.githubusercontent.com/65487373/101979163-e1537b80-3c5a-11eb-9262-8a1575ac6c81.png)
![Screenshot_2020-12-12_01-46-50](https://user-images.githubusercontent.com/65487373/101979167-e57f9900-3c5a-11eb-841b-a57fd6d940fa.png)

**Comment by chiendarret** [comment]
Yes, following
conda update --all
conda install ProDy
The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    prody-1.11                 |   py37h745909e_1         2.3 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.3 MB

The following NEW packages will be INSTALLED:

  prody              conda-forge/linux-64::prody-1.11-py37h745909e_1


Proceed ([y]/n)? y


Downloading and Extracting Packages
prody-1.11           | 2.3 MB    | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done

**Comment by jamesmkrieger** [comment]
Thanks for the bug report. Looks like updating conda worked, right?

**Comment by chiendarret** [comment]
Hi James
No bugs, simply an updated anaconda3 was needed. ProDy installed and
working well. I tried to generate anm files and plots from MD dcd by using
the prody interface. Fine. However, the vmd interface to ProDy has problems
with psf/dcd files limited to the quantum mechanical part of QMMM
simulations (QMonly.dcd files) carried out with NAMD/ORCA. As it is the
first time I used prody, it might reflect some my fault. Separately, I am
posting the problem to VMD in the remote hypothesis that QMonly psf.dcd
have a non standard format. The error log reads;

>
> Traceback (most recent call last):
>   File "/opt/anaconda3/bin/prody", line 11, in <module>
>     sys.exit(prody_main())
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/apps/prody_apps/__init__.py",
> line 53, in prody_main
>     namespace.func(namespace)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/apps/prody_apps/prody_anm.py",
> line 283, in <lambda>
>     **ns.__dict__))
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/apps/prody_apps/prody_anm.py",
> line 75, in prody_anm
>     anm.calcModes(nmodes)
>   File "/opt/anaconda3/lib/python3.7/site-packages/prody/dynamics/anm.py",
> line 219, in calcModes
>     turbo=turbo, expct_n_zeros=6)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/utilities/eigtools.py",
> line 88, in solveEig
>     values, vectors = _eigh(M, eigvals, turbo)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/utilities/eigtools.py",
> line 41, in _eigh
>     values, vectors = linalg.eigh(M, turbo=turbo, eigvals=eigvals)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/scipy/linalg/decomp.py", line
> 488, in eigh
>     'start={}, end={} is given'.format(n-1, lo, hi))
> ValueError: Requested eigenvalue indices are not valid. Valid range is [0,
> 14] and start <= end, but start=0, end=15 is given
> Traceback (most recent call last):
>   File "/opt/anaconda3/bin/prody", line 11, in <module>
>     sys.exit(prody_main())
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/apps/prody_apps/__init__.py",
> line 53, in prody_main
>     namespace.func(namespace)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/apps/prody_apps/prody_anm.py",
> line 283, in <lambda>
>     **ns.__dict__))
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/apps/prody_apps/prody_anm.py",
> line 75, in prody_anm
>     anm.calcModes(nmodes)
>   File "/opt/anaconda3/lib/python3.7/site-packages/prody/dynamics/anm.py",
> line 219, in calcModes
>     turbo=turbo, expct_n_zeros=6)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/utilities/eigtools.py",
> line 88, in solveEig
>     values, vectors = _eigh(M, eigvals, turbo)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/prody/utilities/eigtools.py",
> line 41, in _eigh
>     values, vectors = linalg.eigh(M, turbo=turbo, eigvals=eigvals)
>   File
> "/opt/anaconda3/lib/python3.7/site-packages/scipy/linalg/decomp.py", line
> 488, in eigh
>     'start={}, end={} is given'.format(n-1, lo, hi))
> ValueError: Requested eigenvalue indices are not valid. Valid range is [0,
> 14] and start <= end, but start=0, end=15 is given
>     while executing
> "exec $::NMWiz::pybin $::NMWiz::prody anm --quiet -s $selstr {*}$args
> "$pdbfn""
>     (procedure "::NMWiz::prodySubmitANMjob" line 42)
>     invoked from within
> "::NMWiz::prodySubmitANMjob"
>     (procedure "::NMWiz::prodySubmitJob" line 22)
>     invoked from within
> "::NMWiz::prodySubmitJob"
>     invoked from within
> ".nmwizprody.submitFrame.prodySubmit invoke"
>     ("uplevel" body line 1)
>     invoked from within
> "uplevel #0 [list $w invoke]"
>     (procedure "tk::ButtonUp" line 22)
>     invoked from within
> "tk::ButtonUp .nmwizprody.submitFrame.prodySubmit"
>     (command bound to event)
>

If you are interested I can send you the psf/dcd

Thanks
francesco

**Comment by jamesmkrieger** [comment]
Great to hear things worked. 

What did you do that gave you the error and have you done something without a psf file that works? You shouldn’t really need one.

Best wishes 
James

**Comment by chiendarret** [comment]
Sorry, I do not understand. I loaded the QMonly.psf and QMonly.,dcd files
(the image was quite correct) to VMD, opened the prody interface and worked
the same way that had been successful with MD psf/pdb files, getting the
error that I posted

francesco

**Comment by jamesmkrieger** [comment]
But you’re trying to do ANM analysis, which needs one initial structure not a DCD trajectory. The problem isn’t with the psf but the lack of pdb I think.

**Comment by chiendarret** [comment]
It worked the way I described with classical molecular dynamics (MD)
4fz0_abridged_box_ion.psf/npt-02.dcd files for the same (whole) system
loaded to VMD, getting 4fz0_abridged_box_ion_psf_anm.pdb and
4fz0_abridged_box_ion_psf_anm.nmd, as well as the required plot that I
saved as  a .ps file. I saw the correlation between the residues on the
plot though i did not analyze that as yet.

I used the defaults on the prody interface, so that, from what you say, I
imagine that the nmd file and the plot refer to the frame indicated by the
defaults

francesco

**Comment by jamesmkrieger** [comment]
Hi Francesco,

That's interesting. I guess I'll have to look into that then. It could be a problem with the DCD not the PSF from that method. I know we have problems with DCD files from CHARMM. Yes, please send me the PSF and DCD files to kriegerj@pitt.edu and a PDB file if you have one too.

Many thanks
James

**Comment by jamesmkrieger** [comment]
I just tried your files and I think the problem may be the small number of residues. The default selection "protein and name CA" comes up with only 5 atoms. When I changed it to just "name CA" and there were then 8 atoms then it worked.

However, you shouldn't use the ANM for such small structures anyway. It is made for large structures with hundreds of residues. 

What could be more worthwhile is PCA analysis of this trajectory, which does work for just the 5 CA atoms of the protein.
